services/stats/stats.go

```diff
@@ -48,6 +49,8 @@ func NewStat(Name string, StatType string) (rStats *RudderStats) {
 
 // NewWriteKeyStat is used to create new writekey specific stat. Writekey is added as one of the tags in this case
 func NewWriteKeyStat(Name string, StatType string, writeKey string) (rStats *RudderStats) {
+	newWriteKeyStatLock.Lock()
```

Comment:
Please synchronize the access to destClientsMap as well.

Reply:
Sure